Model/Route.cs

```diff
@@ -0,0 +1,15 @@
+public class Route
```

Comment:
Thanks for making this submission. It looks like it could work, but it might be cleaner to model this after how we worked with paths in data structures. There, a path (in a graph or a tree) would be just not one node, but a starting node, an ending node, and a collection of one or more edges in between (where each edge would be a pair of adjacent nodes) .

Since "path" and "route" are synonymous, I'd suggest you do the same thing here. Basically refactor the ObservableCollection<Route> GetRoutes() that you define in BusinessLogic as
Route GetRoute(), where the ObservableCollection<Leg> would be moved into the Route class.

I hope this makes sense, ask if it doesn't.